# FIN 4000 — Chapter 10: Arbitrage Pricing Theory and Multifactor Models of Risk and Return
*Pricing risk with more than one factor — APT and multifactor models*

**ANSWER KEY — fully worked, instructor reference**

This notebook is the Python companion to the Chapter 10 Excel template — same data, same formulas, same numbers. Where the Excel workshop has you build a formula in a cell, this notebook has you build it in a line of Python instead.

## Step 1 — Entry Quiz *(3 minutes, individually)*

1. How does APT's starting assumption differ from CAPM's (what does APT rely on instead of investors being mean-variance optimizers)?
2. In a multifactor model, what does each factor beta measure?
3. Why can APT work with more than one source of systematic risk while the basic CAPM uses just one (the market)?

---

In [1]:
import numpy as np

## Step 2 — Worked Example *(10 minutes, instructor-led)*

A stock has a beta of 1.2 on the market factor and 0.5 on an interest-rate factor. The market risk premium is 6% and the interest-rate factor's risk premium is 2%. The risk-free rate is 4%. Find the stock's required return under this two-factor model.

In [2]:
r_f = 0.04
beta_m1, rp_m = 1.2, 0.06
beta_r1, rp_r = 0.5, 0.02

In [3]:
required1 = r_f + beta_m1 * rp_m + beta_r1 * rp_r
print(f'Required return = {required1:.2%}')

Required return = 12.20%


*Required return = 4% + 1.2×6% + 0.5×2% = 4%+7.2%+1% = 12.2%.*

## Step 3 — Pair Work *(12 minutes, with a partner)*

**1.** A second stock has β_market=0.9, β_rate=−0.3 (it does well when rates fall). Find its required return.

In [4]:
beta_m2, beta_r2 = 0.9, -0.3
required2 = r_f + beta_m2 * rp_m + beta_r2 * rp_r
print(f'Required return = {required2:.2%}')

Required return = 8.80%


**2.** Why might a stock have a negative beta on the interest-rate factor? Give a real example (think about industries sensitive to borrowing costs).

*(Conceptual — discuss a real example with your partner.)*

**3.** If you used CAPM (the market factor alone) for the first stock, ignoring the rate factor entirely, would you over- or under-estimate its required return, given β_rate=0.5 and a positive rate risk premium?

In [5]:
capm_only1 = r_f + beta_m1 * rp_m
print(f'CAPM-only estimate={capm_only1:.2%}  vs. full 2-factor={required1:.2%}  (understates by {required1-capm_only1:.2%})')

CAPM-only estimate=11.20%  vs. full 2-factor=12.20%  (understates by 1.00%)


## Step 4 — Python Pass *(7 minutes)*

- The multi-factor required-return formula, built so a third or fourth factor could be added just by adding a column
- Both stocks' required returns computed side by side
- A quick comparison row showing what CAPM alone (market factor only) would have estimated for each stock

In [6]:
def multifactor_required_return(r_f, *factors):
    return r_f + sum(beta * rp for beta, rp in factors)

print(multifactor_required_return(r_f, (beta_m1, rp_m), (beta_r1, rp_r)))
print(multifactor_required_return(r_f, (beta_m2, rp_m), (beta_r2, rp_r)))

0.122
0.088


## Step 5 — Discussion *(3-5 minutes, no calculation)*

> Fama and French's three-factor model adds size (SMB) and value (HML) factors to the market factor. Why might a stock's exposure to these factors be compensated with extra expected return, if CAPM says only market risk matters?